### DISCUSSION TITLE
[Office Hours] Retroactive Observations not showing up

### DISCUSSION THREAD

**Discussion by jennxu23** [CS342/.github#38] [Office Hours] [answered]
### Related Discussion

When I try to add observations (specifically step count) that are backdated, they do not get sent to the firebase and do not show up as observations in the mock upload tab.

### Reproduction

1. I start the emulator.
2. I go through the onboarding process (make an account, allow healthkit access).
3. I go to the Health app.
4. I manually add step count data. I pick a random value for the step count, and I change the date to be 1-2 days ago.
5. I navigate back to the app and the mock upload tab to see if the data has been uploaded.
6. I go to the emulator to see if the observation has been sent to the mock firebase.

### Additional context

This feature is more important for debugging, as I want to simulate data being collected throughout a week to test data visualizations. I'm not sure how this issue would manifest in an actual use case, as I assume passive data like step count is usually updated in real time.

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by PSchmiedmayer** [comment, marked as answer]
Hi @jennxu23,

Happy to provide some help with your question and the issue you are describing.

The behavior your are describing is expected, and what we want the default behavior in the CardinalKit HealthKit Module to be like:
The first time an application gets permission to access HealthKit data, it stores the current data as a starting date. All data that is available in the metric with a start date after that date is automatically queried and reported to the application. Any values added before that are not retrieved. This is generally done so for two reasons:
1. ... as we do not generally want to retrieve health data that was collected before the user gave consent to the application.
2. ... as a user might have a lot of data in their Health App (years long), and requesting all that previous data at once might overload most systems and might not be desirable.

Nevertheless, there are use cases where an application might want to collect data retroactively and also gets the user's consent to do so. We currently don't support that feature, but we would be happy to collect what requirements you would have!

Do you need this for your project to support a specific feature, and do you generally need to collect data retrospectively? If so, would you like to provide a hard-coded start date, or do you want this to be dynamic, e.g., 14 days before the user provides consent?

Regarding your comment about debugging: An alternative for debugging would be changing the system clock (on your mac), then restarting the simulator (quit the simulator app and open it again) and therefore setting the iOS simulator to a different time (e.g., in the future) to test the behavior of the app in the future.

**Reply by jennxu23** [reply]
Hi Paul!

Thanks for your response! The actual app doesn't need to collect retroactive data at all -- I only need it for debugging purposes. I'll try your tip for setting my system clock.

**Reply by PSchmiedmayer** [reply]
Sounds great! 🚀

**Reply by jennxu23** [reply]
Hi Paul, 

Following up on this -- I tried various combinations of retroactively adding observations and changing my system's date. The issue I'm running into is that even if there are observations that are backdated (i.e observations collected over three days), the app only collects data from the current date. I think this is just a consequence of having to restart the XCode simulation to reset the date, because the data shows up in the emulator but is not caught by the FHIR API. I would appreciate any suggestions on how to test this functionality (querying data from multiple days) before changing the code too much.
